### PR TITLE
Cleanup JSON instances for Conway governance

### PR DESCRIPTION
**Pull request by lehins**
# Description

This is a collection of small JSON related issues. 

Also addition of Assertions for EPOCH rule that verify that withdrawals and treasury fields of EnactState are always zero before and after the EPOCH rule.

# Checklist

- [x] Commit sequence broadly makes sense and commits have useful messages
- [ ] New tests are added if needed and existing tests are updated
- [ ] When applicable, versions are updated in `.cabal` and `CHANGELOG.md` files according to the
      [versioning process](https://github.com/input-output-hk/cardano-ledger/blob/master/RELEASING.md#versioning-process).
- [ ] The version bounds in `.cabal` files for all affected packages are updated. **If you change the bounds in a cabal file, that package itself must have a version increase.** (See [RELEASING.md](https://github.com/input-output-hk/cardano-ledger/blob/master/RELEASING.md#versioning-process))
- [x] All visible changes are prepended to the latest section of a `CHANGELOG.md` for the affected packages. **New section is never added with the code changes.** (See [RELEASING.md](https://github.com/input-output-hk/cardano-ledger/blob/master/RELEASING.md#changelogmd))
- [x] Code is formatted with [`fourmolu`](https://github.com/fourmolu/fourmolu) (use `scripts/fourmolize.sh`)
- [x] Cabal files are formatted (use `scripts/cabal-format.sh`)
- [x] [`hie.yaml`](https://github.com/input-output-hk/cardano-ledger/blob/master/hie.yaml) has been updated (use `scripts/gen-hie.sh`)
- [x] Self-reviewed the diff

### CODE REVIEW

File: eras/conway/impl/CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,17 @@
 
 ## 1.10.0.0
 
+* Add `psDRepDistrG`.
+* Rename `ensPParams` to `ensCurPParams`.
+* Add `ToJSON` instance for `RatifyState`
+* Change `ToJSON` instance for `ConwayGovState`:
+  * Add `"nextRatifyState"` field
```
Comment (teodanciu) [inline eras/conway/impl/CHANGELOG.md:9]: The backtick + double quote looks a little weird, is it on purpose?  (it doesn't matter too much anyway, just curious)

Comment (lehins) [reply eras/conway/impl/CHANGELOG.md:9]: Those are json fields, so they are represented as strings. Backticks will result in formatting as code, which json is, eg:
image

Comment (teodanciu) [reply eras/conway/impl/CHANGELOG.md:9]: Makes sense, thanks

**Review by teodanciu — APPROVED** [review]
Looks good to me

**Pull request merged**